Added the English version of the docs for asynchronous replication (#9609)
ydb-platform/ydb: ydb/docs/en/core/concepts/async-replication.md

# Asynchronous replication

Asynchronous replication allows synchronizing data between {{ ydb-short-name }} [databases](glossary.md#database) in near real time. Asynchronous replication can also be used for data migration between databases with minimal downtime for applications that work with these databases. Such databases can be located in the same {{ ydb-short-name }} [cluster](glossary.md#cluster) as well as in different clusters.

## Overview {#how-it-works}

Asynchronous replication is based on [Change Data Capture](cdc.md) and deals with logical data. The following diagram shows the replication process:

```mermaid
sequenceDiagram
  participant dst as Source
  participant src as Target

  dst-->src: Initialization
  dst->>dst: Creating an asynchronous replication instance

>   dst->>src: Creating streams of changes

Reviewer: ```suggestion
  dst->>src: Creating changefeeds
```

2. **Target**. A database in which an [asynchronous replication instance](glossary.md#async-replication-instance) and [replicas](glossary.md#replica-object) will be created.

Asynchronous replication consists of the following stages:

* [Initialization](#init).
* [Initial table scan](#initial-scan).
* [Change data replication](#replication-of-changes).

### Initialization {#init}

Initialization of asynchronous replication includes the following:

* Creating an asynchronous replication instance on the target database by using the [CREATE ASYNC REPLICATION](../yql/reference/syntax/create-async-replication.md) YQL expression.
* Establishing a connection with the source database. To connect with the source, the target database uses the [connection parameters](../yql/reference/syntax/create-async-replication.md#params) specified when the  asynchronous replication instance was created.

{% note info %}

The user account that is used to connect to the source database must have the following [permissions](../security/short-access-control-notation.md#access-rights):

* Read permissions for schema objects and directory objects

> * Create, update, delete, and read permissions for streams of changes

Reviewer: ```suggestion
* Create, update, delete, and read permissions for changefeeds
```

{% endnote %}

* The following objects are created for replication objects on the source:

>   * [streams of changes](glossary.md#changefeed) on the source

Reviewer: ```suggestion
  * [changefeeds](glossary.md#changefeed) on the source
```

  * [replicas](glossary.md#replica-object) on the target

{% note info %}

Replicas are created under the user account that was used to create the asynchronous replication instance.

{% endnote %}

### Initial table scan {#initial-scan}

> During the [initial table scan](cdc.md#initial-scan) the source data is exported to streams of changes. The target runs [consumers](topic.md#consumer) that read the source data from the streams of changes and write it to replicas.

Reviewer: ```suggestion
During the [initial table scan](cdc.md#initial-scan) the source data is exported to changefeeds. The target runs [consumers](topic.md#consumer) that read the source data from the changefeeds and write it to replicas.
```

{% endnote %}

For more information about error classes and how to address them, refer to [Error handling](../reference/ydb-sdk/error_handling.md).

## Asynchronous replication completion {#done}

Completion of asynchronous replication might be an end goal of data migration from one database to another. In this case the client stops writing data to the source, waits for the zero replication lag, and completes replication. After the replication process is completed, replicas become available both for reading and writing. Then you can switch the load from the source database to the target database and complete data migration.

{% note info %}

You cannot resume completed asynchronous replication.

{% endnote %}

{% note warning %}

{{ ydb-short-name }} currently supports only **forced** completion of asynchronous replication, when no additional checks are performed for data consistency, replication lag, etc.

{% endnote %}

To complete asynchronous replication, use the [ALTER ASYNC REPLICATION](../yql/reference/syntax/alter-async-replication.md) YQL expression.

## Dropping an asynchronous replication instance {#drop}

When you drop an asynchronous replication instance:

> * Streams of changes are deleted in the source tables.

Reviewer: ```suggestion
* Changefeeds are deleted in the source tables.
```
